src/main/io/gps.c

```diff
@@ -1544,9 +2061,10 @@ static bool _new_speed;
 //15:17:55  R -> UBX NAV,  Size 100,  'Navigation'
 //15:17:55  R -> UBX NAV-SVINFO,  Size 328,  'Satellite Status and Information'
 
-// from the UBlox9 document, the largest payout we receive is the NAV-SAT and the payload size
+// from the UBlox9 document, the largest payload we receive is NAV-SAT, which
 // is calculated as 8 + 12*numCh.  numCh in the case of a M9N is 42.
-#define UBLOX_PAYLOAD_SIZE (8 + 12 * 42)
+// max reported sats can be up to 56
+#define UBLOX_PAYLOAD_SIZE (8 + 12 * GPS_SV_MAXSATS_M8N)
```

Comment:
or use `sizeof` after defining `_buffer`. It will use longest member automatically.
Also, buffer may be shorter than longest message after recent changes

Reply:
Confirm, 100 seems to be the max size with current state.

Reply:
Yes, buffer will be shorter, whenever more than 32 satellites are in view.
Our code needs to handle that gracefully, and not stop us handling the satellite data that we do receive, even when the satellites in view are more than 32 and the received data is truncated.

Reply:
@ledvinap I reverted the recent changes.